src/mkdocstrings/handlers/base.py

```diff
@@ -386,6 +400,28 @@ def __init__(self, config: dict) -> None:
         """
         self._config = config
         self._handlers: Dict[str, BaseHandler] = {}
+        self._url_map = {}
+
+    def register_anchor(self, page: str, anchor: str):
+        """
+        Register that an anchor corresponding to an identifier was encountered when rendering the page.
+
+        Arguments:
+            page: The URL of the current page.
+            anchor: The HTML anchor (without '#') as a string
+        """
+        self._url_map[anchor] = f"{page}#{anchor}"
+
+    def __getitem__(self, identifier: str) -> str:
```

Comment:
I understand it makes it work transparently in the `fix_ref` methods, but I think it would be more readable with a real name like `get_url`. Otherwise it's a bit weird to get an URL when doing `handlers.get("identifier")`, especially when the documentation of the Handlers class is "A collection of handlers" (not a collection of URLs or something :smile_cat: ).

Reply:
Definitely agree, was being lazy (minimizing the code diff)

Reply:
Pushed a commit to address this